### DISCUSSION TITLE
Question on local deployment code changes sync

### DISCUSSION THREAD

**Discussion by ghost** [airbytehq/airbyte#36054] [Connector Questions]
Hi, airbyte/docker newbie here. I've deployed airbyte on my local machine (ubuntu) using docker, and am playing around with connections. If I want to make some changes in airbyte code in my local machine, how do I update my deployment so that it can read the new changed files that are present only at my local machine?

Precisely, I'm making some changes in file `airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py` and want to update my deployment so it starts reading this.

I tried to find some docs for this but couldn't seem to get my head on terminologies used (complete newbie here). Can someone help?

**Comment by pinkasey** [comment]
I'm facing the exact same problem.
Amazing that with so much documentation, and some guides, there's still no explanation of how to develop a connector locally.  

I've read this:
https://github.com/airbytehq/airbyte/blob/master/docs/deploying-airbyte/local-deployment.md
this:
https://github.com/airbytehq/airbyte/blob/master/docs/deploying-airbyte/docker-compose.md
and this (I'm trying to fix a python connector):
https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/cdk-python/README.md
Nothing worked.

**Reply by pinkasey** [reply]
Of course, short time after I posted last comment - I've found where I was wrong.

I was editing https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py  
and I read the instructions in its [README.md](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-google-analytics-v4/README.md), about how to develop it (I'm working on an [older version of it](https://github.com/airbytehq/airbyte/blob/e55e09be8f38c01ae9523801e069656e6bd9944c/airbyte-integrations/connectors/source-google-analytics-v4/README.md), the file has changed since).
I ran this:
`docker build . -t airbyte/source-google-analytics-v4:dev`
and I didn't see my changes (including logging) taking effect.
It took me a while to realize that the connector I have set-up was on version 0.2.0, and I was pushing to docker an image with tag `dev`!
So I deleted my docker images (now I had both `0.2.0`, and `dev`), and ran this:  
`docker build . -t airbyte/source-google-analytics-v4:0.2.0`
and now my changes are taking effect!

[a comment by an automated account is not shown]
